astropy/table/table.py

```diff
@@ -2875,25 +2875,30 @@ def argsort(self, keys=None, kind=None, reverse=False):
         if keys is not None:
             index = get_index(self, names=keys)
             if index is not None:
-                return index.sorted_data()
+                idx = np.asarray(index.sorted_data())
```

Comment:
There was also API inconsistency previously, since `index.sorted_data()` can return a `Column` or `list`, in contrast to the non-indexed case which is an `ndarray`.